app/agents/unified/agent_sparrow.py

```
@@ -1281,10 +1302,59 @@ async def _reserve_model_slot(model_name: str) -> bool:
+                    if fallback_model and fallback_model != runtime.model:
+                        logger.warning(
+                            "gemini_quota_fallback",
+                            primary=runtime.model,
+                            fallback=fallback_model,
+                            error=str(exc),
+                        )
+                        if limiter and reserved_slots:
+                            for model_name, token_identifier in reserved_slots:
+                                try:
+                                    await limiter.release_slot(model_name, token_identifier)
+                                except Exception as release_exc:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)